aws/resource_aws_cloud9_environment_ec2.go

```diff
@@ -0,0 +1,217 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloud9"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloud9EnvironmentEc2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloud9EnvironmentEc2Create,
+		Read:   resourceAwsCloud9EnvironmentEc2Read,
+		Update: resourceAwsCloud9EnvironmentEc2Update,
+		Delete: resourceAwsCloud9EnvironmentEc2Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"automatic_stop_time_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"owner_arn": {
```

Comment:
Nitpick: We can add validation for this attribute via: `ValidateFunc: validateArn,`